rclcpp/src/rclcpp/context.cpp

```diff
@@ -34,8 +35,31 @@ static std::vector<std::weak_ptr<rclcpp::Context>> g_contexts;
 
 using rclcpp::Context;
 
+static
+std::shared_ptr<std::mutex>
+get_global_logging_configure_mutex()
+{
+  static auto mutex = std::make_shared<std::mutex>();
```

Comment:
Because of this, I think it should be possible to create a Context without initializing logging. At least then a user could have their own globals and avoid the global shutdown issues if they manage to init/shutdown logging themselves.

As-is, logging is always configured unconditionally.

Reply:
Probably an option in `rclcpp::InitOptions` and don't get the global mutex in the constructor, instead acquire it conditionally in the `init()` method.

Reply:
@wjwwood @ivanpauno 

maybe it is better for user application to get the flexibility via rclcpp::InitOption, i do agree on this very much.